From 0d2bca57088e9024760c5b0629e7b2df021a4020 Mon Sep 17 00:00:00 2001
From: Zach Reizner <zachr@google.com>
Date: Wed, 23 Jan 2019 14:48:27 -0800
Subject: [PATCH] Revert "[CMake] Unconditionally add .h and .td files to
 target sources"

This reverts commit 6fb010f388bb2cb2f00fe039123092308ac4865d.

Reason for revert: LLVM includes /usr/include/demangle.h which is not installed.

BUG=chromium:924647
TEST=cros tryjob chromiumos-sdk-tryjob
---
 src/llvm/cmake/modules/LLVMProcessSources.cmake | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/src/llvm/cmake/modules/LLVMProcessSources.cmake b/src/llvm/cmake/modules/LLVMProcessSources.cmake
index 7cbd2863500..f65f31d797c 100644
--- a/src/llvm-project/llvm/cmake/modules/LLVMProcessSources.cmake
+++ b/src/llvm-project/llvm/cmake/modules/LLVMProcessSources.cmake
@@ -52,15 +52,16 @@ function(llvm_process_sources OUT_VAR)
   cmake_parse_arguments(ARG "" "" "ADDITIONAL_HEADERS;ADDITIONAL_HEADER_DIRS" ${ARGN})
   set(sources ${ARG_UNPARSED_ARGUMENTS})
   llvm_check_source_file_list( ${sources} )
-  
-  # This adds .td and .h files to the Visual Studio solution:
-  add_td_sources(sources)
-  find_all_header_files(hdrs "${ARG_ADDITIONAL_HEADER_DIRS}")
-  if (hdrs)
-    set_source_files_properties(${hdrs} PROPERTIES HEADER_FILE_ONLY ON)
+  if( LLVM_ENABLE_IDE )
+    # This adds .td and .h files to the Visual Studio solution:
+    add_td_sources(sources)
+    find_all_header_files(hdrs "${ARG_ADDITIONAL_HEADER_DIRS}")
+    if (hdrs)
+      set_source_files_properties(${hdrs} PROPERTIES HEADER_FILE_ONLY ON)
+    endif()
+    set_source_files_properties(${ARG_ADDITIONAL_HEADERS} PROPERTIES HEADER_FILE_ONLY ON)
+    list(APPEND sources ${ARG_ADDITIONAL_HEADERS} ${hdrs})
   endif()
-  set_source_files_properties(${ARG_ADDITIONAL_HEADERS} PROPERTIES HEADER_FILE_ONLY ON)
-  list(APPEND sources ${ARG_ADDITIONAL_HEADERS} ${hdrs})
 
   set( ${OUT_VAR} ${sources} PARENT_SCOPE )
 endfunction(llvm_process_sources)
-- 
2.20.1.321.g9e740568ce-goog
